### PR TITLE
release-23.1: issues: use prefix match to check for existing issue

### PR DESCRIPTION
**Pull request by rafiss**
Backport 1/1 commits from #107444.

/cc @cockroachdb/release

---

It is conventional to annotate a test failure issue title with some additional information about the failure cause. The previous logic was too strict in requiring an exact title match. Now it will match as long as the title matches exactly OR if the prefix matches and is followed by whitespace.

Example:
I added a suffix to [`sql/tests: TestRandomSyntaxFunctions failed [panic in request_job_execution_details]`](https://github.com/cockroachdb/cockroach/issues/106970)  since the issue was with a function the jobs team added. But the next day #107237 was opened anew. This PR would prevent that from happening.

Release justification: non production code change
Epic: None
Release note: None

### CODE REVIEW

File: pkg/cmd/internal/issues/issues.go
Comment (renatolabs) [inline pkg/cmd/internal/issues/issues.go:505]: It just occurred to me: shouldn't we `regexp.QuoteMeta(expectedTitle)`? I don't _think_ this is an immediate problem given our test names and title function, but could become a silent bug if those were to change. What's worse, `MustCompile` would crash the entire roachtest nightly run.

**Review by renatolabs — APPROVED** [review]
Approving, but we should probably address the comment on master. Your call if you want to merge this backport and  then open another one, or wait for that fix to merge and have a single PR.

File: pkg/cmd/internal/issues/issues.go
Comment (rafiss) [reply pkg/cmd/internal/issues/issues.go:505]: yeah that's definitely a good idea

**Comment by rafiss** [comment]
Created https://github.com/cockroachdb/cockroach/pull/107574 on master

**Pull request merged**
